tests/core/policies/test_rule_policy.py

```diff
@@ -92,6 +92,138 @@ def _form_activation_rule(
     )
 
 
+def test_potential_contradiction_resolved_by_conversation_start():
+    """
+    Two rules that contradict each other except that one of them applies only at conversation
+    start -> ensure that this isn't flagged as a contradiction.
+    """
+    utter_anti_greet_action = "utter_anti_greet"
+    domain = Domain.from_yaml(
+        f"""
+intents:
+- {GREET_INTENT_NAME}
+actions:
+- {UTTER_GREET_ACTION}
+- {utter_anti_greet_action}
+    """
+    )
+    policy = RulePolicy()
+    greet_rule_at_conversation_start = TrackerWithCachedStates.from_events(
+        "greet rule at conversation start",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": GREET_INTENT_NAME}),
+            ActionExecuted(UTTER_GREET_ACTION),
+        ],
+    )
+    greet_rule_at_conversation_start.is_rule_tracker = True
+
+    anti_greet_rule = TrackerWithCachedStates.from_events(
+        "anti greet rule",
+        domain=domain,
+        slots=domain.slots,
+        evts=[
+            ActionExecuted(RULE_SNIPPET_ACTION_NAME),
+            ActionExecuted(ACTION_LISTEN_NAME),
+            UserUttered(intent={"name": GREET_INTENT_NAME}),
+            ActionExecuted(utter_anti_greet_action),
+        ],
+    )
+    anti_greet_rule.is_rule_tracker = True
+
+    policy.train(
```

Comment:
could you please add comment that by calling `policy.train` we check that no contradiction is raised

Reply:
Sure.